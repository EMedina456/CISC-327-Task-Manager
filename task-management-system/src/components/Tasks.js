// Program Intention: Handle the view and selection of tasks
// Input/Output: Handle the selection of tasks
// Run Intention: Run with the entire website

// Import files and dependencies here
import { doc, getDoc } from 'firebase/firestore';
import { db } from '../firebase/firebase';
import { onAuthStateChanged, getAuth } from 'firebase/auth';
import { useState, useEffect } from 'react';

const Tasks = ({ handleViewTask, tasks }) => {
  const [taskList, setTaskList] = useState([]);

  async function getTaskInfo() {
    try {
      const auth = getAuth();
      onAuthStateChanged(auth, async (user) => {
        if (user) {
          const uid = user.uid;
          const userRef = doc(db, 'users', uid);
          const docSnap = await getDoc(userRef);
          if (docSnap.exists()) {
            const taskPromises = tasks.map(async (key) => {
              const taskDocRef = doc(db, 'tasks', key);
              const taskDocSnap = await getDoc(taskDocRef);
              if (taskDocSnap.exists()) {
                setTaskList((prevNames) => ({
                  ...prevNames,
                  [key]: taskDocSnap.data().name,
                }));
              }
            });

            await Promise.all(taskPromises);
          } else {
            console.log('No such document!');
          }
        } else {
          window.location.href = '/login';
        }
      });
    } catch (error) {
      console.log(error);
    }
  }

  useEffect(() => {
    getTaskInfo();
  }, []);
  // Tasks Page
  return (
    <div>
      {/* Tasks Page */}
      {/* Be able to click on a task and view it */}
      {taskList.map((key) => (
        <button
          onClick={() => handleViewTask(key)}
          key={key}
          className="flex text-base font-bold mt-2 underline decoration-[#0acdff] md:text-lg lg:text-2xl ">
          {key}
        </button>
      ))}
      <button
        onClick={handleViewTask}
        className="flex text-base font-bold mt-2 underline decoration-[#0acdff] md:text-lg lg:text-2xl ">
        Task 1
      </button>
      <button
        onClick={handleViewTask}
        className="flex text-base font-bold mt-2 underline decoration-[#0acdff] md:text-lg lg:text-2xl ">
        Task 2
      </button>
      <button
        onClick={handleViewTask}
        className="flex text-base font-bold mt-2 underline decoration-[#0acdff] md:text-lg lg:text-2xl ">
        Task 3
      </button>
      <div className="my-16" />
    </div>
  );
};
export default Tasks;
