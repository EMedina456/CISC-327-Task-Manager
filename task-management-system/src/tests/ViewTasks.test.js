/** @jest-environment jsdom */
// Program Intention: Implement Testing for the View Tasks Functionality
// Input/Output: Handle the viewing of tasks
// Run Intention: Run with the other test cases

// Import files and dependencies here
import renderer from 'react-test-renderer'
import Tasks from '../components/Tasks'

// View Tasks Test
it('View Tasks renders correctly', () => {
  // Create a snapshot of the Tasks page
  const tree = renderer.create(<Tasks />).toJSON()
  // Compare the snapshot
  expect(tree).toMatchSnapshot()
})
