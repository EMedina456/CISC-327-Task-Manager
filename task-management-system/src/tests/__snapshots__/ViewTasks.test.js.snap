// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Tasks View Tasks renders correctly 1`] = `
<div>
  <button
    className="flex text-base font-bold mt-2 underline decoration-[#0acdff] md:text-lg lg:text-2xl "
    onClick={[Function]}
  />
  <button
    className="flex text-base font-bold mt-2 underline decoration-[#0acdff] md:text-lg lg:text-2xl "
    onClick={[Function]}
  />
  <button
    className="flex text-base font-bold mt-2 underline decoration-[#0acdff] md:text-lg lg:text-2xl "
    onClick={[Function]}
  />
</div>
`;
